### PR TITLE
[Static references] LPS-182621 Remove static references in module document-library

### PR DESCRIPTION
**Pull request by GabrielHenriquedev**

### CODE REVIEW

**Comment by GabrielHenriquedev** [comment]
ci:test:document

**Comment by catarineaguiar** [comment]
ci:test:dl-store

**Comment by Kvale1733** [comment]
ci:test:dl-store

**Comment by Kvale1733** [comment]
ci:test:document

**Comment by catarineaguiar** [comment]
Some store tests were untested, but I checked that the failure happened before on acceptance, so no regressions were found.

**Comment by tinatian** [comment]
@GabrielHenriquedev @rafaprax For  https://github.com/liferay-core-infra/liferay-portal/pull/4403/commits/a3567ddbece6a76b164851ff48548887fb8ae4f1, DLFileEntryTypeIconProviderUtil is only used in MenuItemProvider which is already a component, please inline the logic into MenuItemProvider

File: modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/criterion/DLItemSelectorCriterionCreationMenuRestrictionUtil.java [hunk 1]
```diff
@@ -64,8 +62,20 @@ public static Set<String> getAllowedCreationMenuUIItemKeys(
 		return allowedCreationMenuUIItemKeys;
 	}
 
-	@Activate
-	protected void activate(BundleContext bundleContext) {
+	private static final PropertyServiceReferenceMapper
+		<String, DLItemSelectorCriterionCreationMenuRestriction>
+			_propertyServiceReferenceMapper =
+				new PropertyServiceReferenceMapper<>("model.class.name");
```
Comment (tinatian) [inline modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/criterion/DLItemSelectorCriterionCreationMenuRestrictionUtil.java:68]: @GabrielHenriquedev @rafaprax Please inline this mapper, no need to make it an attribute of this class

**Comment by tinatian** [comment]
@GabrielHenriquedev @rafaprax please see my comments and resend.

Thanks.
Tina

**Pull request closed without merging**